src/pages/AddItem.js

```diff
@@ -1,3 +1,61 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+// import { useCollection } from 'react-firebase-hooks/firestore';
+
 export default function AddItem(props) {
-  return <h1>This is the form</h1>;
+  const [itemName, setItemName] = useState('');
+  const [purchaseFrequency, setPurchaseFrequency] = useState(0);
+  const [lastPurchased] = useState(null);
+
+  const handleName = (e) => {
+    setItemName(e.target.value);
+  };
+
+  const handleFrequency = (e) => {
+    setPurchaseFrequency(e.target.value);
+  };
+
+  const createListItem = (e) => {
+    e.preventDefault();
+    db.collection('generated_token')
+      .add({
+        item_name: itemName,
+        purchase_frequency: parseInt(purchaseFrequency),
+        last_purchased: lastPurchased,
+      })
+      .then((documentReference) => {
+        console.log('document reference ID', documentReference.id);
+      })
+      .catch((error) => {
+        console.log(error.message);
+      });
+  };
+
+  return (
+    <>
+      <h1>Add Item</h1>
+      <form onSubmit={(e) => createListItem(e)}>
+        <label>
+          Item Name
+          <input
+            type="text"
+            name="name"
+            placeholder="Item Name"
+            value={itemName}
+            onChange={(e) => handleName(e)}
+          />
```

Comment:
Labels for input elements needs to be labeled accessibly for screen readers.
```suggestion
        <label htmlFor="itemName">
          Item Name
          <input
            type="text"
            id="itemName"
            name="name"
            placeholder="Item Name"
            value={itemName}
            onChange={handleName}
          />
```

Reply:
More on  [Accessible Form](https://reactjs.org/docs/accessibility.html).